horqrux/primitives/primitive.py

```diff
@@ -50,8 +52,15 @@ def __post_init__(self) -> None:
         else:
             self.control = Primitive.parse_idx(self.control)
 
+    def __call__(self, state: State | None = None, values: dict[str, Array] = dict()) -> State:
+        from horqrux.apply import apply_gates
+
+        if state is None:
+            state = zero_state(len(self.qubit_support))
+        return apply_gates(state, self, values)
+
     def __iter__(self) -> Iterable:
-        return iter((self.generator_name, self.target, self.control, self.noise, self.sparse))
+        return iter((self,))
```

Comment:
Nice. Would

```suggestion
        return iter(self)
```

work ?

Reply:
No